examples/rest-angular/src/main.ts

```diff
@@ -0,0 +1,15 @@
+import { enableProdMode } from '@angular/core';
+import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+
+import { AppModule } from './app/app.module';
+import { environment } from './environments/environment';
+
+if (environment.production) {
+  enableProdMode();
+} else {
+  import('./mocks/browser').then(x => x.worker.start());
```

Comment:
What is the benefit of making this import async? I'd be careful not to introduce a race condition between importing a worker module and bootstrapping the app. 

Other examples use plain `require`:

```js
if (environment.production) {
 // ...
} else {
  const { worker } = require('./mocks/browser')
  worker.start()
}
```

Reply:
FWIW, I didn't encounter problems with making it async.
I did it because that way I didn't have to include the node typings in order to use `require`.